exec/java-exec/src/main/java/org/apache/drill/exec/record/VectorContainer.java

```diff
@@ -42,6 +42,8 @@
   private final BufferAllocator allocator;
   protected final List<VectorWrapper<?>> wrappers = Lists.newArrayList();
   private BatchSchema schema;
+  private SelectionVector2 selectionVector2;
+  private SelectionVector4 selectionVector4;
```

Comment:
I don't think we should include SV2/SV4 inside a VectorContainer. They are more of RecordBatch level members. VectorContainer should only have metadata and set of vectors in it which is what it is today.

Reply:
This has actually been a point of confusion for me. VectorContainer and RecordBatch both implement VectorAccessible, which has the following methods:

```
  SelectionVector2 getSelectionVector2();
  SelectionVector4 getSelectionVector4();
```

If the intention is that VectorContainers should never hold selection vectors, then the interfaces VectorContainer implements should reflect that. What are you're thoughts about refactoring the VectorAccessable to remove the selection vector methods? In the meantime I will remove the these changes from VectorContainer.

Reply:
Cool made https://issues.apache.org/jira/browse/DRILL-6683 to track that change.